com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -537,6 +537,43 @@ public void UnregisterSideChannel(SideChannel sideChannel)
             }
         }
 
+        /// <summary>
+        /// Returns the SideChannel of Type T if there is one registered, or null if it doesn't.
+        /// If there are multiple SideChannels of the same type registered, only the first found will be returned.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public T GetSideChannel<T>() where T: SideChannel
+        {
+            foreach (var sc in m_SideChannels.Values)
```

Comment:
if there are multiple side channels of the same type but with different GUIDs, what happens?

Reply:
Oops, I updated the comments on the Academy method, but forgot to update here (and ICommunicator). The Academy ones now say:
> /// If there are multiple SideChannels of the same type registered, the returned instance is arbitrary.

Realistically, it'll be the first one that we find.

What does MonoBehavior.GetComponent<>() do if you search for a type that has multiple instances attached? I assume it does something similar. I guess we could keep searching and warn if we find another matching instance...

Reply:
yeah, it actually returns the first one from the order of what you see in the inspector.  As long as it's called out in docs.